src/api_server/src/http.rs

```diff
@@ -115,3 +118,99 @@ pub(crate) async fn attest(
 
     HttpResponse::Unauthorized().json(error_info)
 }
+
+/// GET /resource/{repository}/{type}/{tag}
+/// GET /resource/{type}/{tag}
+pub(crate) async fn resource(
+    request: HttpRequest,
+    repo_dir_path: web::Data<PathBuf>,
+    map: web::Data<SessionMap<'_>>,
+) -> HttpResponse {
+    let error_info = if let Some(cookie) = request.cookie(KBS_SESSION_ID) {
+        if let Some(locked_session) = map.sessions.read().await.get(cookie.value()) {
+            let session = locked_session.lock().await;
+
+            log::info!("Cookie {} request to get resource", session.id());
+
+            if !session.is_expired() {
+                if session.is_authenticated() {
+                    let mut resource_path =
+                        PathBuf::from(repo_dir_path.into_inner().to_str().unwrap_or_default());
+
+                    let resource_repo = request.match_info().get("repository").unwrap_or("default");
+                    let resource_type = request.match_info().get("type").unwrap();
+                    let resource_tag = request.match_info().get("tag").unwrap();
+
+                    if resource_type == "token" {
+                        // TODO: Distribute attestation token (Passport).
+                        return HttpResponse::NotFound()
+                            .message_body(BoxBody::new(
+                                "Token resource is unsupported now".to_string(),
+                            ))
+                            .unwrap();
+                    }
+
+                    log::info!(
+                        "Resource repository: {}, Resource type: {}, Resource tag: {}",
+                        resource_repo,
+                        resource_type,
+                        resource_tag
+                    );
+
+                    // Resource file path:
+                    // `/opt/confidential-containers/kbs/repository/{repository}/{type}/{tag}`
+                    let ref_resource_path =
+                        format!("{resource_repo}/{resource_type}/{resource_tag}");
+                    resource_path.push(ref_resource_path);
+
+                    match ::std::fs::read(&resource_path) {
```

Comment:
Would it be a better choice to define a trait probably like
```rust
trait XXX {
    async fn index_resource(path: &str) -> Result<Vec<u8>>;
}
```

And impl the local fs as this PR, to leave extensibility for other potential storage backends

Reply:
That's definitely a good idea, but I'd propose to leave that for a separate PR?

Reply:
> That's definitely a good idea, but I'd propose to leave that for a separate PR?

Agreed!